lib/grunt-jsonlint-task.js

```diff
@@ -26,70 +26,65 @@ module.exports = function makeTask(grunt, jsonlint, validator, sorter) {
     const options = this.options({
       schema: {},
       ignoreComments: false,
+      ignoreTrailingCommas: false,
       allowSingleQuotedStrings: false,
+      allowDuplicateObjectKeys: true,
       cjson: false,
+      mode: 'json',
       format: false,
       indent: 2,
       sortKeys: false,
       formatter: 'prose',
       reporter: 'exception'
     });
     const { schema } = options;
+    // Share the same options for parsing both data and schema for simplicity.
+    const parserOptions = {
+      mode: options.mode,
+      ignoreComments: options.ignoreComments || options.cjson
+                      || options.mode === 'cjson' || options.mode === 'json5',
+      ignoreTrailingCommas: options.ignoreTrailingCommas || options.mode === 'json5',
+      allowSingleQuotedStrings: options.allowSingleQuotedStrings || options.mode === 'json5',
+      allowDuplicateObjectKeys: options.allowDuplicateObjectKeys,
+      environment: schema.environment
+    };
 
     const format = FORMATTERS[options.formatter];
     const report = REPORTERS[options.reporter];
 
-    if (this.filesSrc != null) {
+    if (this.filesSrc) {
       let failed = 0;
-      let errorDetails = null;
-
-      /* eslint no-param-reassign:
-        ["error", { "props": true, "ignorePropertyModificationsFor": ["jsonlint"] }]
-      */
-
-      // Monkey patch the parseError function to something we can control.
-      const originalParseError = jsonlint.parser.yy.parseError;
```

Comment:
Excellent!  I'm glad you were able to get rid of this old approach for over-riding `parseErrror`.